storm-core/src/jvm/backtype/storm/task/ShellBolt.java

```diff
@@ -376,15 +373,18 @@ public void run() {
                         sendHeartbeatFlag.compareAndSet(true, false);
                     }
 
-                    Object write = _pendingWrites.poll(1, SECONDS);
-                    if (write instanceof BoltMsg) {
-                        _process.writeBoltMsg((BoltMsg) write);
-                    } else if (write instanceof List<?>) {
-                        _process.writeTaskIds((List<Integer>)write);
-                    } else if (write != null) {
-                        throw new RuntimeException("Unknown class type to write: " + write.getClass().getName());
+                    List<Integer> taskIds = _pendingTaskIds.peek();
```

Comment:
Is there any way that we can make this thread block?  The way this is written if the bolt is idle we will be in a tight loop doing nothing but burring through CPU.